markets/retrievaladapter/provider.go

```diff
@@ -61,13 +64,20 @@ func (rpn *retrievalProviderNode) UnsealSector(ctx context.Context, sectorID abi
 		ProofType: si.SectorType,
 	}
 
+	// Set up a pipe so that data can be written from the unsealing process
+	// into the reader returned by this function
 	r, w := io.Pipe()
 	go func() {
 		var commD cid.Cid
 		if si.CommD != nil {
 			commD = *si.CommD
 		}
+		// Unseal the piece into the pipe's writer
```

Comment:
Technically this doesn't always unseal, if we have the unsealed copy around, we'll just read that